### PR TITLE
RMET-593 Barcode Plugin ::: iOS implementation

### PR DESCRIPTION
**Pull request by andregrillo**
## Description

First implementation of the Barcode Plugin for iOS using ZXing library with custom native views

## Context

Closes: https://outsystemsrd.atlassian.net/browse/RMET-593

## Type of changes

- [ ] Fix (non-breaking change which fixes an issue)
- [X] Feature (non-breaking change which adds functionality)
- [ ] Refactor (cosmetic changes)
- [ ] Breaking change (change that would cause existing functionality to not work as expected)

## Platforms affected
- [ ] Android
- [x] iOS
- [ ] JavaScript

## Checklist

- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [ ] Code follows code style of this project
- [ ] CHANGELOG.md file is correctly updated
- [x] Changes require an update to the documentation
- [ ] Documentation has been updated accordingly

### CODE REVIEW

File: src/ios/OSBarcodeScanner.m [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
 
-    if (echo != nil && [echo length] > 0) {
-        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:echo];
+-(void)notificationHandler:(NSNotification *)notice{
+    NSString *str = [notice object];
+    CDVPluginResult* pluginResult = nil;
+    if (str != nil && [str length] > 0 && !([str isEqual: @"error"])) {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:str];
     } else {
         pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
```
Comment (vmfo22) [inline src/ios/OSBarcodeScanner.m:29]: Does it make sense to have a reason for the error?

**Review by vmfo22 — COMMENTED** [review]
Great work @andregrillo! 
Check my questions. Where are you sending the scan result to the plugin?

**Review by CarlsCorrea — APPROVED** [review]

File: src/ios/OSBarcodeScanner.m (hunk 1 above)
Comment (GFOutsystems) [reply src/ios/OSBarcodeScanner.m]: @vmfo22 on this I reviewed and yes. I changed to a a `CDVCommandStatus_NO_RESULT` it was a better response.

File: plugin.xml [hunk 2]
```diff
@@ -36,7 +36,6 @@
     <header-file src="src/ios/ScannerViewController.h"/>
     <source-file src="src/ios/ScannerViewController.xib"/>
 
-    <framework src="ZXingObjC" type="podspec" spec="~> 3.6.4"/>
+    <framework src="https://github.com/OutSystems/ZXingObjC.git" type="podspec" spec="~> 3.6.4"/>
```
Comment (CarlsCorrea) [inline plugin.xml:39]: In this case, is it not necessary to create an outsystems branch with a 3.6.4-OS release?

Comment (GFOutsystems) [reply plugin.xml:39]: That is actually a good question. I don't really know. But I think its by the TAG.

**Review by alexgerardojacinto — APPROVED** [review]

File: plugin.xml (hunk 2 above)
Comment (GFOutsystems) [reply plugin.xml]: This changed @CarlsCorrea I can confirm that the Tag works.

**Pull request merged**